src/Transport/Topology/TopologyCreator.cs

```diff
@@ -52,6 +72,14 @@ public TopologyCreator(ITransportPartsContainer container, IManageNamespaceManag
             }
         }
 
+        public async Task AssertManagedRights()
+        {
+            if (!await hasManageRights)
```

Comment:
Scratch the comment. This uses a custom awaiter which doesn't require ConfigureAwait(false)